rasterio/_base.pyx

```diff
@@ -763,6 +764,20 @@ cdef class DatasetBase(object):
                 stacklevel=2)
             return Affine.from_gdal(*self.get_transform())
 
+    def subdatasets(self):
+        """Return a list of subdatasets"""
+        tags = self.tags(ns='SUBDATASETS')
+        subs = defaultdict(dict)
+        for key, val in tags.items():
+            _, idx, fld = key.split('_')
+            fld = fld.lower()
+            if fld == 'desc':
+                fld = 'description'
```

Comment:
My comment about `$ rio info --subdatasets` applies here too.  If `GDALGetDescription()` produces the description then it might be more useful to have this property just return a list of subdataset connection strings.

Reply:
True. I'll do that if possible.

Reply:
Bad news: GDAL doesn't surface subdataset metadata in GDALGetDescription.